### PR TITLE
Refactor extract features

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, the `extract-features` module is refactored to include CP features (as well as the DP features that were already included). 

In this module it is necessary to create two envs (one for each feature extraction type) as well as two bash files to run each of these extraction processes.

### CODE REVIEW

File: 2.extract-features/CP-feature-extraction/extract-cp-features.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+# path to CellProfiler pipeline
+pipeline_path = pathlib.Path("process-cell-health.cppipe")
+# iterate through plates in load path
+for plate_path in nuclei_images_load_path.iterdir():
+    print(f"Extracting CP features for plate {plate_path.name}...")
+
+    # iterate through image paths in plates (there should only be one)
+    for image_folder_path in plate_path.iterdir():
```
Comment (d33bs) [inline 2.extract-features/CP-feature-extraction/extract-cp-features.py]: Where does `image_folder_path` get used? Consider linking this variable to the loop somehow.

File: 2.extract-features/CP-feature-extraction/extract-cp-features.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        features_folder_path.mkdir(exist_ok=True, parents=True)
+
+        # file to log CP output
+        log_file_path = pathlib.Path(
+            f"{features_save_path}/{plate_path.name}/CP_plate_run.log"
+        )
+
+        # extract features for particular plate
+        with open(
+            log_file_path,
```
Comment (d33bs) [inline 2.extract-features/CP-feature-extraction/extract-cp-features.py]: Thinking about the potential of the log file here and not knowing what the command below might do, consider providing a start, end, and duration timestamp to help indicate when things occurred.

File: 2.extract-features/CP-feature-extraction/process-cell-health.cpproj
Comment (d33bs) [inline 2.extract-features/CP-feature-extraction/process-cell-health.cpproj:1]: How does this file get created in relation to the similar process-cell-health.cppipe file? Are these linked in some way that could be defined within either file (so as to show this work within source control)? If this is unreasonable, please feel free to disregard.

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py [hunk 3]
```diff
@@ -165,20 +165,20 @@ def compile_training_locations(
...
             )
 
             # handle errors for no locations file/no data
             try:
                 frame_segmentations = pd.read_csv(
                     frame_segmentations_path, delimiter="\t"
                 )
-            except:
-                print(f"No segmentation data for {frame_segmentations_path.name}")
+            except Exception as e:
```
Comment (d33bs) [inline 2.extract-features/DP-feature-extraction/DP-project-utils.py:176]: Consider using a more specific exception here to provide greater description of what the expectation is for this `except` branch.

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py [hunk 4]
```diff
@@ -165,20 +165,20 @@ def compile_training_locations(
...
+                print(f"No segmentation data for {frame_segmentations_path}")
```
Comment (d33bs) [inline 2.extract-features/DP-feature-extraction/DP-project-utils.py]: Consider making use of `e` here to provide greater detail on the exception encountered. Alternatively, if it's unhelpful, consider removing the definition for `as e` above.

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py [hunk 5]
```diff
@@ -213,19 +213,19 @@ def compile_project(
     # make project dir
     project_path.mkdir(parents=True, exist_ok=True)
     
-    # copy necessary DP files from DP_files/ to DP project
-    config_name = f"cell_health_{object}_config.json"
-    copy_DP_files(project_path, config_name, checkpoint_name)
+    # # copy necessary DP files from DP_files/ to DP project
+    # config_name = f"cell_health_{object}_config.json"
+    # copy_DP_files(project_path, config_name, checkpoint_name)
```
Comment (d33bs) [inline 2.extract-features/DP-feature-extraction/DP-project-utils.py]: For any of the commented lines here or below, consider removing them if they are no longer needed. Alternatively, consider providing a comment about the commented lines to generally describe why they were retained.

File: 2.extract-features/README.md [hunk 6]
```diff
@@ -2,9 +2,49 @@
 
 In this module, we present our pipeline for extracting features from the Cell Health data.
 
-### Feature Extraction
+# CellProfiler Feature Extraction
```
Comment (d33bs) [inline 2.extract-features/README.md]: Consider making this a header-2 (`##`) as there is already a top-level header defined. The general guidance here stems from HTML, where it is recommended to use only header-1 per document ([see here for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#avoid_using_multiple_h1_elements_on_one_page)).

File: 2.extract-features/README.md [hunk 7]
```diff
@@ -2,9 +2,49 @@
...
+
+```bash
+# Run this script to extract CP features
+bash 2a.extract-cp-features.sh
+```
+
+**Note**: [extract-cp-features.ipynb](CP-feature-extraction/extract-cp-features.ipynb) does not show all output as we ran this code elsewhere.
+On our machine, this pipeline took about 12 hours per plate (108 hours total).
+
+# DeepProfiler Feature Extraction
```
Comment (d33bs) [inline 2.extract-features/README.md]: Consider making this a header-2 (similar comment to the above).

File: 3.preprocess-features/preprocess-features.ipynb
Comment (d33bs) [inline 3.preprocess-features/preprocess-features.ipynb:1]: Noticing the change here, should there be a related python file update?

File: 2.extract-features/2.cp-feature-extraction-env.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+name: 2.cp-feature-extraction-cell-health
+
+channels:
+  - conda-forge
+
+dependencies:
+  - conda-forge::python=3.8.13
```
Comment (d33bs) [inline 2.extract-features/2.cp-feature-extraction-env.yml]: Could Python 3.8.16 be used here (preferring later updates)?

File: 2.extract-features/2.cp-feature-extraction-env.yml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+name: 2.cp-feature-extraction-cell-health
```
Comment (d33bs) [inline 2.extract-features/2.cp-feature-extraction-env.yml:1]: Consider using the project name as a prefix for the environment name which might assist with the ordering when viewing many projects and environments within the same system. For example: `cell-health-data_2.cp-feature-extraction` (or similar).

**Review by d33bs — APPROVED** [review]
Nice job! I left a few minor comments and suggestions, overall this LGTM! Please don't hesitate to let me know if you have any questions.

File: 2.extract-features/CP-feature-extraction/extract-cp-features.py (hunk 1 above)
Comment (roshankern) [reply 2.extract-features/CP-feature-extraction/extract-cp-features.py]: That variable is unused, great catch! It has been removed.

File: 2.extract-features/CP-feature-extraction/process-cell-health.cpproj
Comment (roshankern) [reply 2.extract-features/CP-feature-extraction/process-cell-health.cpproj:1]: This file is actually no longer used in this module (I forgot to remove it after testing). Thanks for the catch, it has been removed!

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py (hunk 3 above)
Comment (roshankern) [reply 2.extract-features/DP-feature-extraction/DP-project-utils.py:176]: Great suggestion. I have changed this code to show the exception that has occurred (`e`). This way, the user can track the error that has occurred and the compilation will not stop for the exception.

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py (hunk 4 above)
Comment (roshankern) [reply 2.extract-features/DP-feature-extraction/DP-project-utils.py]: Just did this, thanks for the suggestion!

File: 2.extract-features/DP-feature-extraction/DP-project-utils.py (hunk 5 above)
Comment (roshankern) [reply 2.extract-features/DP-feature-extraction/DP-project-utils.py]: Oops, none of this code should be commented (it shouldn't have changed at all during this commit). Forgot to fix that after testing, great catch!

File: 3.preprocess-features/preprocess-features.ipynb
Comment (roshankern) [reply 3.preprocess-features/preprocess-features.ipynb:1]: There should be but it probably makes more sense to make both of these changes when I refactor the next module... I'll revert the change here.

File: 2.extract-features/2.cp-feature-extraction-env.yml (hunk 8 above)
Comment (roshankern) [reply 2.extract-features/2.cp-feature-extraction-env.yml]: Yep, implemented!

File: 2.extract-features/2.cp-feature-extraction-env.yml (hunk 9 above)
Comment (roshankern) [reply 2.extract-features/2.cp-feature-extraction-env.yml:1]: I think that change would be best left for another PR as I am following the environment name precedent set in the previous modules of this repo.

File: 2.extract-features/CP-feature-extraction/extract-cp-features.py (hunk 2 above)
Comment (roshankern) [reply 2.extract-features/CP-feature-extraction/extract-cp-features.py]: Good idea, included!

**Pull request merged**